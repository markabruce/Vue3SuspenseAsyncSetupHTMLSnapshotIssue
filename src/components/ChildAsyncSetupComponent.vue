<template>
<!-- eslint-disable max-len -->
  <div>This is the Async Setup Component</div>
  <!-- eslint-enable max-len -->
</template>

<script>
export default {
  async setup() {
    const someFunctionThatReturnsAPromise = () => new Promise((resolve) => {
      setTimeout(() => {
        resolve();
      }, 1000);
    });

    await someFunctionThatReturnsAPromise();
  },
};
</script>

<style>
</style>
