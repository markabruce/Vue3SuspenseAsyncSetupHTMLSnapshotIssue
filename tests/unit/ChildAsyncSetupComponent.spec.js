import { mount, flushPromises } from '@vue/test-utils'
import { defineComponent, nextTick } from 'vue'
import ChildAsyncSetupComponent from '@/components/ChildAsyncSetupComponent.vue'

describe('ChildAsyncSetupComponent.vue', () => {
  let wrapper;

  beforeEach(async () => {
    const ChildAsyncSetupComponentWrappedInSuspense = defineComponent({
      components: { ChildAsyncSetupComponent },
      template: '<Suspense><ChildAsyncSetupComponent/></Suspense>'
    })
    
    wrapper = mount(ChildAsyncSetupComponentWrappedInSuspense);
    
    await flushPromises();
    await nextTick();
  });

  it('renders to match html snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  })
})
