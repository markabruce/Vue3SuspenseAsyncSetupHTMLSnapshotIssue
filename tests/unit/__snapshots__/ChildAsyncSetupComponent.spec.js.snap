// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChildAsyncSetupComponent.vue renders to match html snapshot 1`] = `""`;
